src/components/Picklist/index.js

```diff
@@ -303,11 +235,14 @@ class Picklist extends Component {
             onFocus,
             name,
         } = this.props;
-        const { context, activeOptionName } = this.state;
+        const { activeOptionName } = this.state;
         const ariaLabel = title || assistiveText;
         const { label, icon, name: currentValueName } = this.getValue();
         const providerContext = {
-            ...context,
+            privateOnClick: this.handleOptionClick,
+            privateRegisterChild: this.registerChild,
+            privateUnregisterChild: this.unregisterChild,
+            privateOnHover: this.hoverChild,
```

Comment:
what I meant (we can put this in a class variable like this.context, see above) is that it is not necessary in state but you can put it in constructor like:
```
this.context = {
                privateOnClick: this.handleOptionClick,
                privateRegisterChild: this.registerChild,
                privateUnregisterChild: this.unregisterChild,
                privateOnHover: this.hoverChild,
            }
```

Reply:
this is just a waste of memory.... as in the render we include activeOptionName and  currentValueName...  that are dynamic variables... there is no need to have this twice... another option is to create a function buildContext() and pass it directly to Provider value.. i think its less confusing